lib/ts/recipe/totp/types.ts

```diff
@@ -0,0 +1,143 @@
+        | { status: "OK" | "INVALID_TOTP_ERROR" | "TOTP_NOT_ENABLED_ERROR" }
+        | { status: "LIMIT_REACHED_ERROR"; retryAfterMs: number }
+    >;
+    verifyDevice: (input: {
+        userId: string;
+        deviceName: string;
+        totp: string;
+        userContext: any;
+    }) => Promise<
+        | { status: "OK" | "INVALID_TOTP_ERROR" | "TOTP_NOT_ENABLED_ERROR" | "UNKNOWN_DEVICE_ERROR" }
+        | { status: "LIMIT_REACHED_ERROR"; retryAfterMs: number }
+    >;
+    updateDevice: (input: {
+        userId: string;
+        existingDeviceName: string;
+        newDeviceName: string;
+        userContext: any;
+    }) => Promise<{ status: "OK" | "TOTP_NOT_ENABLED_ERROR" | "DEVICE_ALREADY_EXISTS_ERROR" | "UNKNOWN_DEVICE_ERROR" }>;
+    removeDevice: (input: {
+        userId: string;
+        deviceName: string;
+        userContext: any;
+    }) => Promise<{ status: "OK" | "TOTP_NOT_ENABLED_ERROR" }>;
+    listDevices: (input: {
+        userId: string;
+        userContext: any;
+    }) => Promise<
+        | { status: "OK"; devices: { name: string; period: number; skew: number; verified: boolean }[] }
+        | { status: "TOTP_NOT_ENABLED_ERROR" }
+    >;
+};
+
+export type APIOptions = {
+    recipeImplementation: RecipeInterface;
+    appInfo: NormalisedAppinfo;
+    config: TypeNormalisedInput;
+    recipeId: string;
+    isInServerlessEnv: boolean;
+    req: BaseRequest;
+    res: BaseResponse;
+};
+
+export type APIInterface = {
+    createDevicePOST?: (input: {
+        session: SessionContainer;
+        deviceName: string;
+        period: number;
+        skew: number;
+        options: APIOptions;
+        userContext: any;
+    }) => Promise<{ status: "OK"; qr: string } | { status: "DEVICE_ALREADY_EXISTS_ERROR" }>; // TODO: Consider seperately mentioning templates vars used in qr (issuerName, secret, provisioner, etc)
+    verifyCodePOST?: (input: {
+        session: SessionContainer;
+        totp: string;
+        options: APIOptions;
+        userContext: any;
+    }) => Promise<
+        | { status: "OK" | "INVALID_TOTP_ERROR" | "TOTP_NOT_ENABLED_ERROR" }
+        | { status: "LIMIT_REACHED_ERROR"; retryAfterMs: number }
+    >;
+    verifyDevicePOST?: (input: {
+        session: SessionContainer;
+        deviceName: string;
+        totp: string;
+        options: APIOptions;
+        userContext: any;
+    }) => Promise<
+        | { status: "OK" | "INVALID_TOTP_ERROR" | "TOTP_NOT_ENABLED_ERROR" | "UNKNOWN_DEVICE_ERROR" }
+        | { status: "LIMIT_REACHED_ERROR"; retryAfterMs: number }
+    >;
+    removeDevicePOST?: (input: {
+        session: SessionContainer;
+        deviceName: string;
+        options: APIOptions;
+        userContext: any;
+    }) => Promise<{ status: "OK" | "TOTP_NOT_ENABLED_ERROR" }>;
+    listDevicesGET?: (input: {
+        session: SessionContainer;
+        options: APIOptions;
+        userContext: any;
+    }) => Promise<{ status: "OK" | "TOTP_NOT_ENABLED_ERROR" }>;
+};
```

Comment:
You need to add API for checking if totp is setup for a user - or can the frontend know this based on the output of listDevicesGET (if this returns TOTP_NOT_ENABLED_ERROR, then it means it's not enabled, else it is)

Reply:
Yes they can know it from that. But we can still create an API.

Reply:
Good point. Removed the API

Reply:
Not sure what do you mean. I removed the `isTotpEnabled` API. 

See 

- https://github.com/supertokens/supertokens-node/blob/8fde4a6a105abf945efb5e33dffa4b0f0b4362db/lib/ts/recipe/totp/api/implementation.ts#L20
- https://github.com/supertokens/supertokens-node/blob/8fde4a6a105abf945efb5e33dffa4b0f0b4362db/lib/ts/recipe/totp/types.ts#L111